bpf/lib/nat.h

```diff
@@ -1021,10 +1016,9 @@ snat_v4_rev_nat(struct __ctx_buff *ctx, const struct ipv4_nat_target *target,
 #ifdef ENABLE_SCTP
 	case IPPROTO_SCTP:
 #endif  /* ENABLE_SCTP */
-		if (ctx_load_bytes(ctx, off, &l4hdr, sizeof(l4hdr)) < 0)
+		if (ipv4_ct_extract_l4_ports(ctx, off, CT_INGRESS, &tuple, &has_l4_header) < 0)
```

Comment:
Hm ideally we would pass the `ip4` through, and not re-validate it internally. But that's probably just a follow-on cleanup?

On the other hand could also just open-code the fragment look-up, same as `lb4_extract_tuple()`.

Reply:
Staged that cleanup here: https://github.com/cilium/cilium/pull/28717

I left it in draft for now, your call if we want to merge your PR first or rebase :).

Reply:
:merged:.